src/Microsoft.VisualStudio.Editors/OptionPages/GeneralOptions.vb

```diff
@@ -32,8 +42,83 @@ Namespace Microsoft.VisualStudio.Editors.OptionPages
             End Set
         End Property
 
+        ''' <summary>
+        ''' This setting has dynamic default value:
+        '''     - true if VS was installed from Preview channel
+        '''     - false if VS was installed from Release channel.
+        ''' When default setting value is changed by user, we write current value to a marker file 
+        ''' located under local AppData VS instance folder to be used by out-proc components.
+        ''' If there no marker file there, it measn VS is having default setting value which was 
+        ''' never changed by user manually. Out-proc components should treat absense of the file as 
+        ''' default option value and check the channel using same setup API we use here. 
+        ''' This absense=default logic is needed for the case VS is installed, was never started yet,
+        ''' but user runs msbuild form dev command line to build solution. In this case msbuild 
+        ''' components should have a way to figure out default setting on their own.
+        ''' 
+        ''' Note: This setting is not roamed. After chatting with shell team, we decided not to roam 
+        ''' it due to stability issues.
+        ''' </summary>
+        ''' <returns></returns>
+        Public Property UsePreviewSdk As Boolean
+            Get
+                Return If(_settingsManager?.GetValueOrDefault(UsePreviewSdkSettingKey, _isPreview), _isPreview)
+            End Get
+            Set
+                _settingsManager.SetValueAsync(UsePreviewSdkSettingKey, Value, isMachineLocal:=False)
```

Comment:
I checked with Huy from shell team and we looked up the code that is using this option. It seems that isMahcineLocal is not related to roaming, it is determines if the setting would be shared on the machine across different SKUs and potentially instances even though we did not find how exactly it is reused since it is being stored to registry storage which is stored under instance folders.  We want False in this case, just to be safe.

---
In reply to: [192156768](https://github.com/dotnet/project-system/pull/3602#discussion_r192156768) [](ancestors = 192156768)